okta/okta.go

```diff
@@ -43,14 +46,27 @@ type resource struct {
 	client *Client
 }
 
-func NewClient(config *Config) *Client {
+func NewClient(config *Config, httpClient *http.Client, cacheManager cache.Cache) *Client {
```

Comment:
This feels like a good opportunity to take `okta.client.requestTimeout` and apply it as `httpClient.Timeout`.  It also looks like we can take `okta.client.connectionTimeout` and use that as the idle timeout via a `Transport`:

```
tr := &http.Transport{
	IdleConnTimeout:    30 * time.Second
}
client := &http.Client{Transport: tr}
```